src/odemis/driver/hamamatsurx.py

```diff
@@ -256,12 +252,17 @@ def _setBinning(self, value):
         """
         # ResolutionVA need tuple instead of list of format [2 x 2]
         binning = "%s x %s" % (value[0], value[1])
+
+        # If camera is acquiring, it is essential to stop cam first and then change binning.
+        # Currently, this only affects the Alignment tab, where camera is continuously acquiring.
+        self.parent.AcqStop()  # TODO check with HW!
```

Comment:
You probably don't really want to start live acquisition every time the binning is changed? Only if the acquisition was already playing before hand. 
Ensuring proper thread safety might not be easy (eg, the binning is changed just before acquiring). That's another reason prefer to delay the actual binning change to the next acquisition (via update_settings()).

So I'd consider this current solution as a "band-aid", and put a `TODO use update_settings() instead`.

Reply:
yes I also wasn't happy with this solution. How would I check that an acquisition is already running? The thread "self.t_image" will be always active correct? How about a variable which is set True when starting an acquisition, which can be checked if True/False before calling update_settings()?

Reply:
If you use the update_settings() "pattern", it's easy to know whether the acquisition is running or not, as it'd run from the acquisition thread.

If you want to keep it simple for now, you could have a `.active` boolean attribute on the DataFlow, which is updated in the `start_generate()`/`stop_generate()` functions. (That's what I've already implemented in some other drivers. Note that it's not thread-safe, so in some corner cases it might not report the right state (but as a band-aid, it's fine).